backend/db.js

```diff
@@ -0,0 +1,106 @@
+import { DataSource } from 'apollo-datasource'
+import crypto from 'crypto'
+
+export class Post {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString('hex')
+        Object.assign(this, data)
+    }
+}
+
+export class User {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString('hex')
+        Object.assign(this, data)
+    }
+}
+
+export class InMemoryDataSource extends DataSource {
+    constructor() {
+        super()
+        this.posts = []
+        this.users = []
+        this.initDummyUsers()
+        this.initDummyPosts()
+    }
+
+    initialize(...args) {
+        //Warum wird diese Methode so oft ausgeführt?
```

Comment:
`ApolloServer` calls this on every request, in order to give access to request-specific data like e.g. authentication data. You will need that in exercise `#4`.